### PR TITLE
protonmail-bridge: 3.0.18 -> 3.0.21

### PR DESCRIPTION
**Pull request by MrFreezeex**
###### Description of changes
- Updated protonmail-bridge 3.0.18 -> 3.0.20
- Changed back the binary name from bridge -> protonmail-bridge because it conflicts with iproute2 bridge and it is actually named like that in other deb/rpm packages (not in the Makefile though, which motivated this naming change in the first place).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by drupol — APPROVED** [review]
LGTM !

**Comment by dotlambda** [comment]
Why are there two changelog entries?

**Comment by MrFreezeex** [comment]
> Why are there two changelog entries?

It was added 3 weeks ago here https://github.com/NixOS/nixpkgs/pull/218025 for the bump of a major version and the renaming of the binary which ended up being a bad idea. So now it's only for the bump of the major version

**Comment by MrFreezeex** [comment]
> Why are there two changelog entries?
Ah wait I just actually understood the question 🤦 there is one in notable change and one in breaking change, I might want to remove the one in breaking change now that the binary is no longer renamed though 🤔 🤷‍♂️

**Comment by SuperSandro2000** [comment]
Changelog entries go in either especially if they are the same.

**Comment by MrFreezeex** [comment]
> Changelog entries go in either especially if they are the same.

Oki, my bad then, I dropped the one in breaking change since I don't believe it's that necessary if the binary are no longer renamed.

**Comment by ajmasia** [comment]
Hi guys! Does the app GUI work with this package? I am testing the previous version and I can only use the CLI :pray:

**Comment by MrFreezeex** [comment]
> Hi guys! Does the app GUI work with this package? I am testing the previous version and I can only use the CLI 🙏

Unfortunately this is still CLI only indeed. GUI would probably need more work as AFAIK it consist of two programs the launcher in go without cmake and the actual GUI in c++ with cmake. I don't have a need for the GUI so I am not super interested in doing this TBH.

**Comment by ajmasia** [comment]
> Unfortunately this is still CLI only indeed. GUI would probably need more work as AFAIK it consist of two programs the launcher in go without cmake and the actual GUI in c++ with cmake. I don't have a need for the GUI so I am not super interested in doing this TBH.

Thanks! How do you keep proton-mail running in the background? :blush:

**Comment by drupol** [comment]
I don't think there is a systemd module. Actually, it would be a good idea to have one.

**Comment by FedX-sudo** [comment]
This has fixed some of the bugs I have been having with Protonmail. Thanks!

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 1]
```diff
@@ -134,7 +134,7 @@ In addition to numerous new and upgraded packages, this release has the followin
 
 - The [services.wordpress.sites.&lt;name&gt;.plugins](#opt-services.wordpress.sites._name_.plugins) and [services.wordpress.sites.&lt;name&gt;.themes](#opt-services.wordpress.sites._name_.themes) options have been converted from sets to attribute sets to allow for consumers to specify explicit install paths via attribute name.
 
-- `protonmail-bridge` package has been updated to v3.0 and the CLI executable is now named bridge instead of protonmail-bridge to be more in line with upstream.
+- `protonmail-bridge` package has been updated to v3.0.
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2305.section.md:137]:
```suggestion
- `protonmail-bridge` package has been updated to major version 3.
```

**Comment by SuperSandro2000** [comment]
FTBF fpr aarch64-linux

```
# github.com/ProtonMail/proton-bridge/v3/cmd/Desktop-Bridge
/nix/store/8mkiaqipr8d434kbs0ah0my6k4m002j1-go-1.19.7/share/go/pkg/tool/linux_arm64/link: running g++ failed: exit status 1
/nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lparser: No such file or directory
/nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lparser_gen: No such file or directory
/nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lproto: No such file or directory
/nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lprotobuf: No such file or directory
/nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lantlr4-runtime: No such file or directory
collect2: error: ld returned 1 exit status
error: builder for '/nix/store/7j8rprgdnrq59a3qf8jahkvmsxhxqw0w-protonmail-bridge-3.0.21.drv' failed with exit code 1;
```

**Comment by MrFreezeex** [comment]
> FTBF fpr aarch64-linux
> 
> ```
> # github.com/ProtonMail/proton-bridge/v3/cmd/Desktop-Bridge
> /nix/store/8mkiaqipr8d434kbs0ah0my6k4m002j1-go-1.19.7/share/go/pkg/tool/linux_arm64/link: running g++ failed: exit status 1
> /nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lparser: No such file or directory
> /nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lparser_gen: No such file or directory
> /nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lproto: No such file or directory
> /nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lprotobuf: No such file or directory
> /nix/store/didj9sv4gfv0dgq2k5hrayr4r5jpkfl9-binutils-2.40/bin/ld: cannot find -lantlr4-runtime: No such file or directory
> collect2: error: ld returned 1 exit status
> error: builder for '/nix/store/7j8rprgdnrq59a3qf8jahkvmsxhxqw0w-protonmail-bridge-3.0.21.drv' failed with exit code 1;
> ```

IIUC next version will no longer have c++ embed so it might just works without too much packaging hassle very soon.

**Pull request merged**